cellpack/autopack/Gradient.py

```diff
@@ -91,6 +91,28 @@ def __init__(self, gradient_data):
 
         self.function = self.defaultFunction  # lambda ?
 
+    @staticmethod
+    def update_ingredient_gradient(ingr, arguments):
+        """
+        Update the ingredient gradient
+        """
+        if (
+            "gradient" in arguments
+            and arguments["gradient"] != ""
+            and arguments["gradient"] != "None"
+        ):
```

Comment:
this seems like something that can be cleaned up in the recipe loader (not have multiple falsy possibilities)

Reply:
Good idea! I added some validation and tests to check whether gradient information is entered correctly during ingredient creation from recipe arguments.